plugins/releases/index.js

```diff
@@ -1,90 +1,213 @@
+    if (interval > 1) {
+      return `${interval} months ago`;
+    }
+    if (interval === 1) {
+      return `${interval} month ago`;
+    }
+    interval = Math.floor(seconds / 86400);
+    if (interval > 1) {
+      return `${interval} days ago`;
+    }
+    if (interval === 1) {
+      return `${interval} day ago`;
+    }
+    interval = Math.floor(seconds / 3600);
+    if (interval > 1) {
+      return `${interval} hours ago`;
+    }
+    if (interval === 1) {
+      return `${interval} hour ago`;
+    }
+    interval = Math.floor(seconds / 60);
+    return `${interval} minutes ago`;
+  }
 
 function format_release(release) {
-  return `**• v${release.version} (${release.id}) ${release.current ? "^^^- current^^^" : ""}**
-  ***Build:*** ${release.slug ? release.slug.id : 'unknown'}
-  ***Created:*** ${(new Date(release.created_at)).toLocaleString()}
-  ***Status:*** ${release.status === "succeeded" ? "^^succeeded^^" : "!!" + release.status + "!!"}
-  ***Description:*** ${release.description}\n`;
+  console.assert(release.build, 'No build information was found.')
+  let info = [
+    release.current ? "^^^current^^^" : "", 
+    release.description,
+    release.build.source_blob.author, 
+    release.build.source_blob.message ? `##${release.build.source_blob.message.replace(/#/g, '').replace(/\s+/g, ' ')}##` : '',
+    release.build.source_blob.commit ? `${release.build.source_blob.commit.substring(0, 7)}` : '', 
+  ].filter(x => x && x !== '').map((x) => x.toString().replace(/\n/g, ' '));
+  return `**• v${release.version}**\t${getDateDiff(new Date(release.created_at))}\t${info.join(' - ')}`
 }
 
-function list(appkit, args) {
-  console.assert(args.app && args.app !== '', 'An application name was not provided.');
-  appkit.api.get('/apps/' + args.app + '/releases', 
-    appkit.terminal.format_objects.bind(null, format_release, appkit.terminal.markdown('###===### No releases were found.')));
+async function find_release(appkit, app, release_key) {
+  let get = util.promisify(appkit.api.get)
+  if(/^[\w]{8}-[\w]{4}-[\w]{4}-[\w]{4}-[\w]{12}$/.exec(release_key) !== null) {
+    return await get(`/apps/${app}/releases/${release_key}`)
+  } else if(release_key === 'latest' || release_key === 'current') {
+    let results = await get(`/apps/${app}/releases`)
+    return results[results.length - 1]
+  } else if (/^v[0-9]+$/.exec(release_key) !== null || !Number.isNaN(parseInt(release_key, 10))) {
+    let version = parseInt(release_key, 10)
+    if(Number.isNaN(version)) {
+      version = parseInt(release_key.substring(1), 10)
+    }
+    console.assert(!Number.isNaN(version), 'The version, was not... a version.')
+    let results = await get(`/apps/${app}/releases`)
+    results = results.filter((x) => x.version === version)
+    console.assert(results.length === 1, `The version ${version} was not found.`)
+    return results[0]
+  } else {
+   let results = await get(`/apps/${app}/releases`)
+    return results[results.length - 2]
```

Comment:
It gets the second to last? 

Will this fail if there are < 2?

Reply:
yeah, i've added some error catching and reworked the logic a bit to be more explicit.